### PR TITLE
Convert official build pipeline to follow the 1ES templates

### PR DESCRIPTION
**Pull request by steveisok**
We will need to update the official pipeline post build.

### CODE REVIEW

**Comment by steveisok** [comment]
Test run of official pipeline https://dev.azure.com/dnceng/internal/_build/results?buildId=2401538&view=results

File: eng/azure-pipelines-official.yml [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+                targetPath: $(Build.SourcesDirectory)/artifacts/VSSetup/$(_BuildConfig)/Insertion/
+                artifactName: 'Workloads'
+              continueOnError: true
+              condition: ne(variables['SkipPublishWorkloads'], 'true')
+            - powershell: $(CheckNuGetSizesScript)
+              displayName: Check NuGet sizes
+
+    - template: /eng/common/templates-official/post-build/post-build.yml@self
+      parameters:
+        publishingInfraVersion: 3
```
Comment (akoeplinger) [inline eng/azure-pipelines-official.yml:292]: this is default now so we can remove it

**Review by akoeplinger — APPROVED** [review]

**Comment by vseanreesermsft** [comment]
@akoeplinger 
@steveisok 

This needs to be done for downlevel servicing branches

**Comment by akoeplinger** [comment]
@vseanreesermsft the changes are now backported in 6.0/7.0/8.0 branches :)

**Pull request merged**